### ISSUE TITLE
tensorflow.vectorized_map cannot recive loop variable

### ISSUE THREAD

**Issue by sybwjdnr** [tensorflow/tensorflow#43822]
<em>Please make sure that this is a bug. As per our
[GitHub Policy](https://github.com/tensorflow/tensorflow/blob/master/ISSUES.md),
we only address code/doc bugs, performance issues, feature requests and
build/installation issues on GitHub. tag:bug_template</em>

**System information**
- Have I written custom code (as opposed to using a stock example script provided in TensorFlow):
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): Linux Ubuntu 16.04
- Mobile device (e.g. iPhone 8, Pixel 2, Samsung Galaxy) if the issue happens on mobile device:
- TensorFlow installed from (source or binary): ananconda
- TensorFlow version (use command below): 2.0.2
- Python version:3.7
- Bazel version (if compiling from source):
- GCC/Compiler version (if compiling from source):
- CUDA/cuDNN version:10.2
- GPU model and memory: GTX1080ti RTX2080ti

You can collect some of this information using our environment capture
[script](https://github.com/tensorflow/tensorflow/tree/master/tools/tf_env_collect.sh)
You can also obtain the TensorFlow version with:
1. TF 1.0: `python -c "import tensorflow as tf; print(tf.GIT_VERSION, tf.VERSION)"`
2. TF 2.0: `python -c "import tensorflow as tf; print(tf.version.GIT_VERSION, tf.version.VERSION)"`

**Describe the current behavior**
The tensorflow.vectorized_map function cannot receive the positional parameters in the loop variable.

**Describe the expected behavior**
The tensorflow.vectorized_map function can receive the positional parameters in the loop variable like its predecessor map_fn function.
**Standalone code to reproduce the issue**
Provide a reproducible test case that is the bare minimum necessary to generate
the problem. If possible, please share a link to Colab/Jupyter/any notebook.
arangelist = tf.cast(tf.range(start=0, limit=1000 - 100 + 1, delta=1), dtype=tf.float32)
def errfunc(Index):
    """
    generate input for tf.vectorized_map
    :param Index:
    :return:
    """
    Index = tf.cast(Index, dtype=tf.int32)
    return Index
c = tf.vectorized_map(errfunc, arangelist)

**Other info / logs** Include any logs or source code that would be helpful to
diagnose the problem. If including tracebacks, please include the full
traceback. Large logs and files should be attached.

**Comment by bhack** [comment]
Can you format your code block?

**Comment by sybwjdnr** [comment]
`import tensorflow as tf
arangelist = tf.cast(tf.range(start=0, limit=1000 - 100 + 1, delta=1), dtype=tf.float32)
Temp_Input_Tensor = tf.ones((128, 4000))

def errfunc(Index):
    """

    :param Index:
    :return:
    """
    Index = tf.cast(Index, dtype=tf.int32)
    Sequence_Fragment_Tensor = Temp_Input_Tensor[:,Index * 4:(Index * 4 + 10 * 4)]  ## [batch_size, kernel_length]

    return Sequence_Fragment_Tensor
c = tf.vectorized_map(errfunc, arangelist)
`



if I run the code above, I will get the "ValueError: Input "input" of op "StridedSlice" expected to be not loop invariant"

**Comment by bhack** [comment]
```
import tensorflow as tf
arangelist = tf.cast(tf.range(start=0, limit=1000 - 100 + 1, delta=1), dtype=tf.float32)
Temp_Input_Tensor = tf.ones((128, 4000))

def errfunc(Index):
Index = tf.cast(Index, dtype=tf.int32)
Sequence_Fragment_Tensor = Temp_Input_Tensor[:,Index * 4:(Index * 4 + 10 * 4)]
return Sequence_Fragment_Tensor
c = tf.vectorized_map(errfunc, arangelist)
```

**Comment by bhack** [comment]
With TF 2.3.1 I just get a Warning:
`WARNING:tensorflow:Using a while_loop for converting StridedSlice`

**Comment by amahendrakar** [comment]
@sybwjdnr,
I was able to reproduce the error with [TF v2.0.2](https://colab.research.google.com/gist/amahendrakar/430f6301b99595e38f0ad3e5bdc888ed/43822-2-0.ipynb), however the issues seems to be fixed with [TF v2.3](https://colab.research.google.com/gist/amahendrakar/72e8a4d02125bc75e144cb133796a612/43822.ipynb). Please find the attached gist.

Could you please update TensorFlow to v2.3 and check if it works. Thanks!

**Comment by sybwjdnr** [comment]
thank you!
